Mathlib/Analysis/NormedSpace/Ray.lean

```diff
@@ -54,6 +54,12 @@ theorem norm_smul_eq (h : SameRay ℝ x y) : ‖x‖ • y = ‖y‖ • x := by
 
 end SameRay
 
+lemma norm_nsmul [NormedSpace ℝ E] (n : ℕ) (x : E) : ‖n • x‖ = n • ‖x‖ := by
+  induction' n with n ih
+  · simp
+  · rw [succ_nsmul, succ_nsmul, SameRay.norm_add, ih]
+    exact sameRay_nsmul_right _ _
```

Comment:
This has a more elementary proof that belongs in another file:

```lean
lemma norm_nsmul [NormedAddCommGroup E] [NormedSpace ℝ E] (n : ℕ) (x : E) : ‖n • x‖ = n • ‖x‖ := by
  rw [nsmul_eq_smul_cast ℝ, norm_smul, IsROrC.norm_natCast, nsmul_eq_mul]
```
or even
```lean
lemma norm_nsmul [IsROrC 𝕜] [NormedAddCommGroup E] [NormedSpace 𝕜 E] (n : ℕ) (x : E) : ‖n • x‖ = n • ‖x‖ := by
  rw [nsmul_eq_smul_cast 𝕜, norm_smul, IsROrC.norm_natCast, nsmul_eq_mul]
```
next to `IsROrC.norm_natCast`.

Reply:
Nice!